frontend/src/views/new-project/NewProjectView.tsx

```diff
@@ -60,66 +60,101 @@ const createProjectMutation = gql`
+    const changeNameEvent: ChangeNameEvent = { type: 'CHANGE_NAME', name: value };
+    dispatch(changeNameEvent);
   };
 
   const onCreateNewProject = async (event) => {
     event.preventDefault();
-
     const variables = {
       input: {
         name: name.trim(),
         visibility: 'PUBLIC',
       },
     };
+    const submitEvent: SubmitEvent = { type: 'SUBMIT' };
+    dispatch(submitEvent);
     createProject({ variables });
   };
 
   useEffect(() => {
     if (!loading && !error && data) {
       if (error) {
-        dispatch({ type: HANDLE_SUBMIT_FAILURE__ACTION, reponse: error });
+        let errorMessage = error.message;
+        const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message: errorMessage };
+        dispatch(showToastEvent);
       } else if (data) {
-        dispatch({ type: HANDLE_SUBMIT__ACTION, response: data });
+        let id = data.createProject?.project?.id;
+        if (id) {
+          const HandleSuccessEvent: HandleSuccessEvent = { type: 'HANDLE_SUCCESS', id };
+          dispatch(HandleSuccessEvent);
+        }
+        else if (data.createProject?.message) {
+          const handlErrorEvent: HandleErrorEvent = { type: 'HANDLE_ERROR' };
+          dispatch(handlErrorEvent);
+          dispatch({ type: 'SHOW_TOAST', message: data.createProject?.message });
+        }
       }
     }
-  }, [loading, data, error]);
+  }, [loading, data, error, dispatch]);
 
-  if (newProjectId) {
+  if (newProjectView === 'success') {
     return <Redirect to={`/projects/${newProjectId}/edit`} />;
   }
 
-  const canBeSubmitted = viewState === VALID__STATE;
-  const nameMessageContent = nameMessage ? (
-    <Message severity={nameMessageSeverity} data-testid="nameMessage" content={nameMessage} />
-  ) : null;
+  const canBeSubmitted = newProjectView === 'valid';
+  let validationMessage = '';
+  if (newProjectView === 'invalid') {
+    validationMessage = 'The name must contain between 3 and 20 characters.';
+  }
+
   return (
     <View condensed>
-      <FormContainer title="Create a new project" subtitle="Get started by creating a new project" banner={message}>
+      <FormContainer title="Create a new project" subtitle="Get started by creating a new project">
         <Form onSubmit={onCreateNewProject}>
-          <Label value="Name">
-            <Textfield
-              name="name"
-              placeholder="Enter the project name"
-              value={name}
-              onChange={onNameChange}
-              autoFocus
-              data-testid="name"
-            />
-            <div className={styles.messageArea}>{nameMessageContent}</div>
-          </Label>
-          <Buttons>
-            <ActionButton type="submit" disabled={!canBeSubmitted} label="Create" data-testid="create-project" />
-            <SecondaryButton type="submit" to={`/`} label="Cancel" data-testid="back" />
-          </Buttons>
+          <TextField
+            error={newProjectView === 'invalid'}
```

Comment:
It works because the example is really really small but this logic should be part of the state machine